### PR TITLE
[KIECLOUD-122] - Enable JMS for RHDM and RHPAM images.

### PR DESCRIPTION
**Pull request by spolti**
Follow up changes.
Need to expose the tcp-ssl port, 61617.

Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by tchughesiv** [comment]
@spolti we've gone the route of versioning product configs. i'll be updating the master branch over the next couple of days to account for 7.5.0 and at that time you'll need to rebase and push your changes specifically against only the 7.5.0 config dir. sorry bout that.

**Comment by tchughesiv** [comment]
@spolti rebase once this merges - https://github.com/kiegroup/kie-cloud-operator/pull/237

**Comment by tchughesiv** [comment]
@bmozaffa this PR includes crd changes. 
Given the strong possibility of crd changes with minor releases and the new ask around versioning validation within the CRD. is now the time to start versioning the api with each release?

**Comment by errantepiphany** [comment]
@spolti , When you return from PTO, can you please rebase? Thanks!

**Comment by bmozaffa** [comment]
Yes, assuming they are commonly overwritten. Otherwise they can be provided in env vars. I assume these need to be configured for authoring ha? If so, let's ask @swati-kale to do a separate PR.

**Comment by spolti** [comment]
@tchughesiv updated, and I moved the jms changes from 7.4.1 dir to 7.5.0.

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 1]
```diff
@@ -424,9 +425,57 @@ func TestRhpamProdImmutableJMSEnvironment(t *testing.T) {
...
+
+	assert.Equal(t, "test-jms-rhpamcentrmon", env.Console.DeploymentConfigs[0].ObjectMeta.Name)
+	assert.Equal(t, "test-jms-kieserver", env.Servers[0].DeploymentConfigs[0].Name)
+	assert.Equal(t, "test-jms-kieserver-postgresql", env.Servers[0].DeploymentConfigs[1].Name)
+	assert.Equal(t, "test-jms-kieserver-amq", env.Servers[0].DeploymentConfigs[2].Name)
+	assert.Equal(t, "amq-jolokia-console", env.Servers[0].Routes[1].Name)
+	assert.False(t, env.Servers[0].Routes[1].Spec.TLS == nil)
+	assert.Equal(t, "amq-tcp-ssl", env.Servers[0].Routes[2].Name)
+	assert.False(t, env.Servers[0].Routes[2].Spec.TLS == nil)
+	testAMQEnvs(t, env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env, env.Servers[0].DeploymentConfigs[2].Spec.Template.Spec.Containers[0].Env)
```
Comment (tchughesiv) [inline pkg/controller/kieapp/defaults/defaults_test.go]: can you add a check here that `jms.AMQEnableSSL` is being appropriately set to true?

Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults_test.go]: good point, I forget to add this check.

**Comment by swati-kale** [comment]
> Yes, assuming they are commonly overwritten. Otherwise they can be provided in env vars. I assume these need to be configured for authoring ha? If so, let's ask @swati-kale to do a separate PR.

@spolti  Do you mean  PR for adding new fields in Operator UI for JMS section (line # 389 to 454) wrt https://github.com/kiegroup/kie-cloud-operator/blob/master/deploy/crds/kieapp.crd.yaml#L389?

**Pull request merged**